src/app/clusters/tls-certificate-management-server/tls-certificate-management-server.h

```diff
@@ -0,0 +1,346 @@
+
+    virtual ~TlsCertificateManagementDelegate() = default;
+
+    using RootCertificateListCallback     = std::function<CHIP_ERROR(DataModel::List<const RootCertStructType> & certs)>;
+    using ClientCertificateListCallback   = std::function<CHIP_ERROR(DataModel::List<const ClientCertStructType> & certs)>;
+    using LoadedRootCertificateCallback   = std::function<CHIP_ERROR(RootCertStructType & cert)>;
+    using LoadedClientCertificateCallback = std::function<CHIP_ERROR(ClientCertStructType & certs)>;
+    using GeneratedCsrCallback            = std::function<Protocols::InteractionModel::Status(ClientCsrResponseType & cert)>;
+
+    /**
+     * @brief Appends a root certificate to the provisioned root certificates list maintained by the delegate.
+     *        The delegate must ensure it makes a copy of the provided request and data, if any.
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the certificate is associated with
+     * @param[in] provisionReq The request data specifying the root certificate to be provisioned
+     * @param[out] outCaid a reference to the TLSCAID variable that is to contain the ID of the provisioned root cert.
+     *
+     * @return CHIP_NO_ERROR if the certificate was appended to the list successfully.
+     * @return CHIP_ERROR if there was an error adding the certificate.
+     */
+    virtual Protocols::InteractionModel::ClusterStatusCode ProvisionRootCert(EndpointId matterEndpoint, FabricIndex fabric,
+                                                                             const ProvisionRootCertificateType & provisionReq,
+                                                                             Tls::TLSCAID & outCaid) = 0;
+
+    /**
+     * @brief Executes loadedCallback for each root certificate. The certificate passed to loadedCallback has a guaranteed lifetime
+     * of the method call.
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the certificate is associated with
+     * @param[in] loadedCallback lambda to execute for each certificate.  If this function returns an error result,
+     * iteration stops and returns that same error result.
+     */
+    virtual CHIP_ERROR LoadedRootCerts(EndpointId matterEndpoint, FabricIndex fabric,
+                                       LoadedRootCertificateCallback loadedCallback) const = 0;
+
+    /**
+     * @brief Executes loadedCallback on all matching certificates. The certificates passed to loadedCallback has a guaranteed
+     * lifetime of the method call.
+     *
+     * @param[in] matterEndpoint The matter endpoint to query against
+     * @param[in] fabric The fabric the certificate is associated with
+     * @param[in] loadedCallback lambda to execute on list of all matching certificates.  If this function returns an error result,
+     * iteration stops and returns that same error result.
```

Comment:
What iteration?  The function is called once...

Reply:
iteration of the root certs; per method doc, "Executes loadedCallback on all matching certificates"; iteration stops if `loadedCallback` returns non-success